lib/rules/no-extra-boolean-cast.js

```diff
@@ -47,23 +56,67 @@ module.exports = {
             "ForStatement"
         ];
 
+        /**
+         * Check if a node is a Boolean function or constructor.
+         * @param {ASTNode} node the node
+         * @returns {boolean} If the node is Boolean function or constructor
+         */
+        function isBooleanFunctionOrConstructorCall(node) {
+
+            // Boolean(<bool>) and new Boolean(<bool>)
+            return (node.type === "CallExpression" || node.type === "NewExpression") &&
```

Comment:
Do we want to ensure that `Boolean` isn't a shadowed global variable?

Reply:
That should be also done in `CallExpression(node)`.

Though I think that both global `Boolean` fixes can be a separate PR because it's an existing issue (this function was extracted from `UnaryExpression(node)`).

Reply:
Separating it out into a separate PR sounds good to me!